http/src/request/mod.rs

```diff
@@ -1,7 +1,7 @@
 macro_rules! poll_req {
-    ($ty: ty, $ret: ty) => {
+    ($ty: ty, $out: ty) => {
         impl std::future::Future for $ty {
-            type Output = ::std::result::Result<$ret, $crate::error::Error>;
+            type Output = Result<$crate::request::Response<$out>, $crate::error::Error>;
```

Comment:
should this still be ::std::result::Result?

Reply:
Using `::std::result::Result` is only really required in certain cases and this isn't one of them. It being here in the first place was a mistake on my part.